code/datums/components/crayonwalk.dm

```diff
@@ -0,0 +1,22 @@
+
+/datum/component/wearertargeting/crayonwalk // stonepillar's crayon project
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	signals = list(COMSIG_MOVABLE_MOVED)
+	mobtype = /mob/living/carbon/human
+	proctype = .proc/crayonwalk
+
+/datum/component/wearertargeting/crayonwalk/proc/crayonwalk(mob/living/carbon/human/H, last_turf, direct)
+	if (!H.lying && istype(src.parent, /obj/item/clothing/shoes/clown_shoes))
+		var/obj/item/clothing/shoes/clown_shoes/S = H.shoes
+		if (length(S.crayons))
```

Comment:
Imo this thing shouldn't draw if there's already a crayon drawing present on the tile. Otherwise you're gonna get stuff like 
![image](https://user-images.githubusercontent.com/358431/106905701-d538e700-66fc-11eb-82be-4ed1ec013e9a.png)
A good way for checking that would be
```suggestion
		if (length(S.crayons) && !(locate(/obj/decal/cleanable/writing) in last_turf))
```